src/Symfony/Component/AssetMapper/ImportMap/RemotePackageStorage.php

```diff
@@ -53,9 +58,14 @@ public function save(ImportMapEntry $entry, string $contents): void
 
         $vendorPath = $this->getDownloadPath($entry->packageModuleSpecifier, $entry->type);
 
-        @mkdir(\dirname($vendorPath), 0777, true);
-        if (false === @file_put_contents($vendorPath, $contents)) {
-            throw new RuntimeException(error_get_last()['message'] ?? \sprintf('Failed to write file "%s".', $vendorPath));
+        try {
+            $this->filesystem->dumpFile($vendorPath, $contents);
+        } catch (IOException $e) {
+            throw new RuntimeException(\sprintf('Failed to write file "%s".', $vendorPath), 0, $e);
+        }
+
+        if (!is_file($vendorPath) || !is_writable($vendorPath)) {
```

Comment:
I'm not sure this makes sense: if dumpFile succeeded, why check again here?

Reply:
I'm spiraling around the same problem: I don't find a simple way to test this on both linux & windows .. 

(i'm very tempted to remove the test as it's now coverd by Filesystem tests)

Open to any idea/advice here :)